packages/eslint-plugin/src/rules/require-await.ts

```diff
@@ -63,6 +63,15 @@ export default util.createRule<Options, MessageIds>({
 
         case AST_NODE_TYPES.ArrowFunctionExpression:
           rules.ArrowFunctionExpression(node);
+
+          // If body type is not BlockStatment, we need to check the return type here
+          if (!(node.body.type === 'BlockStatement')) {
```

Comment:
nit - use the enum type, and `!==`
```suggestion
          if (node.body.type !== AST_NODE_TYPES.BlockStatement) {
```